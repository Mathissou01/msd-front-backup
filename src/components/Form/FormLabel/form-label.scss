@use "src/styles/01-settings/settings.colors" as colors;

.c-FormLabel {
  $self: &;

  display: flex;
  flex-direction: column;
  gap: 2px;
  margin-bottom: 2px;
  color: colors.$blue-2;
  font-weight: 700;
  font-size: 16px;
  line-height: 24px;

  &_row {
    flex-direction: row;
    gap: 8px;
    width: 100%;
  }

  &_styleTable {
    color: colors.$white;
    line-height: 20px;
  }

  &__LabelWrapper {
    display: flex;
    flex-direction: row;
    flex-grow: 1;
    gap: 10px;
    align-items: center;
    justify-content: space-between;

    #{$self}_row & {
      flex-wrap: nowrap;
    }
  }

  &__Label {
    margin-right: auto;
  }

  &__LabelDescription {
    margin-left: 8px;
    font-weight: 350;
  }

  &__Secondary,
  &__Validation {
    color: colors.$blue-1;
    font-weight: 400;
    font-family: DINPro-Regular, serif;

    #{$self}_styleTable & {
      color: colors.$blue-4;
      font-weight: 500;
    }
  }

  &__Secondary {
    color: colors.$gray-800;
  }

  &__Validation {
    max-width: 75%;
    font-size: 14px;
    line-height: 21px;
    text-align: right;

    #{$self}_styleTable & {
      font-size: 14px;
    }

    &_left {
      margin-bottom: 4px;
      text-align: left;
    }
  }

  &__Content {
    display: flex;
    flex-direction: row;

    #{$self}_row & {
      flex-grow: 7;
    }
  }
}
