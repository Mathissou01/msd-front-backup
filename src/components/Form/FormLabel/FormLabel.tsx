import classNames from "classnames";
import React from "react";
import "./form-label.scss";

export type LabelStyle = "default" | "table";

export type ValidationStyle = "inline" | "multiline";

export interface IFormLabelProps {
  children?: React.ReactNode;
  label?: string;
  labelDescription?: string;
  secondaryLabel?: string;
  validationLabel?: string;
  forId?: string;
  tagType?: "label" | "legend";
  isRequired?: boolean;
  flexStyle?: "column" | "row";
  labelStyle?: LabelStyle;
  validationStyle?: ValidationStyle;
}

export default function FormLabel({
  children,
  label,
  labelDescription,
  secondaryLabel,
  validationLabel,
  forId,
  tagType = "label",
  isRequired = false,
  flexStyle = "column",
  labelStyle = "default",
  validationStyle = "inline",
}: IFormLabelProps) {
  const Tag = tagType;
  const labelClassNames = classNames("c-FormLabel", {
    "c-FormLabel_row": flexStyle === "row",
    "c-FormLabel_styleTable": labelStyle === "table",
  });

  return (
    <div className={labelClassNames}>
      <Tag className="c-FormLabel__LabelWrapper" htmlFor={forId}>
        <span className="c-FormLabel__Label">
          {`${label}${isRequired ? " *" : ""}`}
          {labelDescription && (
            <span className="c-FormLabel__LabelDescription">
              {labelDescription}
            </span>
          )}
        </span>
        {validationStyle === "inline" &&
          flexStyle === "column" &&
          validationLabel && (
            <span className="c-FormLabel__Validation">{validationLabel}</span>
          )}
      </Tag>
      {validationStyle === "multiline" &&
        flexStyle === "column" &&
        validationLabel && (
          <span className="c-FormLabel__Validation c-FormLabel__Validation_left">
            {validationLabel}
          </span>
        )}
      {secondaryLabel && (
        <Tag className="c-FormLabel__LabelWrapper" htmlFor={forId}>
          <span className="FormLabel__Secondary">{secondaryLabel}</span>
        </Tag>
      )}
      {children && <div className="c-FormLabel__Content">{children}</div>}
      {flexStyle === "row" && (
        <Tag className="c-FormLabel__LabelWrapper" htmlFor={forId}>
          {validationLabel && (
            <span className="c-FormLabel__Validation">{validationLabel}</span>
          )}
        </Tag>
      )}
    </div>
  );
}
