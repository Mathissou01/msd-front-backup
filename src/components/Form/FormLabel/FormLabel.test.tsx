import { render, screen } from "@testing-library/react";
import React from "react";
import FormLabel from "./FormLabel";

const mock = {
  name: "form-label",
  label: "form label",
  secondaryLabel: "secondary label",
  validationLabel: "validation label",
};

describe("FormLabel", () => {
  it("renders", () => {
    const { container } = render(
      <FormLabel
        label={mock.label}
        secondaryLabel={mock.secondaryLabel}
        validationLabel={mock.validationLabel}
        forId={mock.name}
        isRequired={false}
      />,
    );

    const label = screen.getByText("form label");
    expect(label).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
