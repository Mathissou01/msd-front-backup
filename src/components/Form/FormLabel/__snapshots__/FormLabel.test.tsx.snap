// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormLabel renders 1`] = `
<div>
  <div
    class="c-FormLabel"
  >
    <label
      class="c-FormLabel__LabelWrapper"
      for="form-label"
    >
      <span
        class="c-FormLabel__Label"
      >
        form label
      </span>
      <span
        class="c-FormLabel__Validation"
      >
        validation label
      </span>
    </label>
    <label
      class="c-FormLabel__LabelWrapper"
      for="form-label"
    >
      <span
        class="FormLabel__Secondary"
      >
        secondary label
      </span>
    </label>
  </div>
</div>
`;
